lib/rules/define-macros-order.js

```diff
@@ -228,17 +246,28 @@ function create(context) {
   }
 
   /**
-   * @param {ASTNode} tokenBefore
-   * @param {ASTNode} tokenAfter
+   * Get result text to insert
+   * @param {string} textNode
+   * @param {ASTNode} target
    */
-  function getTextBetweenTokens(tokenBefore, tokenAfter) {
-    return sourceCode.text.slice(tokenBefore.range[1], tokenAfter.range[0])
+  function getInsertText(textNode, target) {
+    const afterTargetComment = sourceCode.getTokenAfter(target, {
+      includeComments: true
+    })
+    const afterText = sourceCode.text.slice(
+      target.range[1],
+      afterTargetComment.range[0]
+    )
+    // handle case when a();b() -> b()a();
+    const invalidResult = !textNode.endsWith(';') && !/[\n;]/.test(afterText)
```

Comment:
I guess it should check the start only, right?

```suggestion
    const invalidResult = !textNode.endsWith(';') && !/^[\n;]/.test(afterText)
```

Reply:
No, this value also get valid result
```ts
const afterText = '   ;   '
```

Reply:
But wouldn't it also return true for `b();c()`? Or `(b();)`?

Reply:
The thing here is what is inside afterText. afterText contains text between the node and the next comment. So it can contain only whitespaces and new lines.
While I was writing this I understood that there is no need to look for `;`. There is no change afterText can contain it :)

Reply:
hm

Reply:
maybe i need to remove `;` from regexp

Reply:
Changed it here https://github.com/vuejs/eslint-plugin-vue/pull/1863/files#diff-8293acf6fd19c2affb306c4d724be1279c64123427e207a444b5ecdab4789940R262